clients/cli/src/main/java/org/apache/gravitino/cli/commands/Command.java

```diff
@@ -115,12 +136,26 @@ protected GravitinoAdminClient buildAdminClient() {
       client = client.withVersionCheckDisabled();
     }
     if (authentication != null) {
-      if (authentication.equals("simple")) {
+      if (authentication.equals(SIMPLE_AUTH)) {
         if (userName != null && !userName.isEmpty()) {
           client = client.withSimpleAuth(userName);
         } else {
           client = client.withSimpleAuth();
         }
+      } else if (authentication.equals(OAUTH_AUTH)) {
```

Comment:
This peace of code is duplicated with the buildClient() method; is there a way to reuse?

Reply:
There was no way I coudl find to reuse it I did try. One part of the code uses GravitinoAdminClient and the other GravitinoClient, which are not interchangeable when using the Builder.

Reply:
If you extract the common code and try using it from both methods, you get errors with converting to/from Builder<GravitinoClient> and Builder<GravitinoAdminClient>.

Reply:
Looks like you can work around that by using `Builder<GravitinoClientBase>`, I thought it didn't have all the methods.

Reply:
Nope that doesn't work either, as you get "Builder<GravitinoClient> cannot be converted to Builder<GravitinoClientBase>"

Reply:
finally worked out a way but it's not pretty.